packages/core/src/components/widgets/SelectWidget.tsx

```diff
@@ -67,7 +67,37 @@ function SelectWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extend
   );
 
   const selectedIndexes = enumOptionsIndexForValue<S>(value, enumOptions, multiple);
-  const showPlaceholderOption = !multiple && schema.default === undefined;
+
+  let opts = null
+  if (optgroups) {
+    let enumOptionFromValue = new Map()
+    enumOptions.forEach((enumOption, i) => {
+      enumOptionFromValue.set(enumOption.value, [enumOption, i])
+    })
+    opts = Object.keys(optgroups).map(optgroup => {
+      return (
+        <optgroup key={optgroup} label={optgroup}>{optgroups[optgroup].map(value => {
+          if (!enumOptionFromValue.has(value)) return null
+          const disabled = enumDisabled && enumDisabled.indexOf(value) !== -1;
+          let [{ label }, i] = enumOptionFromValue.get(value)
+          return (
+            <option key={i} value={String(i)} disabled={disabled}>
+              {label}
+            </option>
+          );
+        })}</optgroup>
+      )
+    })
+  } else {
+    opts = Array.isArray(enumOptions) && enumOptions.map(({ value, label }, i) => {
+      const disabled = enumDisabled && enumDisabled.indexOf(value) !== -1;
+      return (
+        <option key={i} value={String(i)} disabled={disabled}>
+          {label}
+        </option>
+      );
```

Comment:
This is duplicate code to what is above. Consider creating a new component at the top of the file to render it and use it here and above

Reply:
Something like a helper component in the file:

```
interface OptionProps {
  enumDisabled?: UIOptionsType['enumDisabled'];
  value: EnumOptionsType['value'];
  label: EnumOptionsType['label'];
  index: number;
}

function Option(props: OptionProps) {
  const { enumDisabled, index, label, value } = props;
const disabled = enumDisabled && enumDisabled.indexOf(value) !== -1;
      return (
        <option value={String(index)} disabled={disabled}>
          {label}
        </option>
      )
}
```